plugin/clone/src/clone_hton.cc

```diff
@@ -105,6 +106,14 @@ static bool run_hton_clone_begin(THD *thd, plugin_ref plugin, void *arg) {
   return (false);
 }
 
+static bool run_collect_expected_storage_vector(THD *, plugin_ref plugin,
+                                                void *arg) {
+  auto clone_arg = static_cast<std::unordered_set<legacy_db_type> *>(arg);
+  auto hton = plugin_data<handlerton *>(plugin);
```

Comment:
I think we should only collect InnoDB and MyRocks plugins, or something that supports transactions (though there is code for blackhole to be transactional). It is possible for other storage engine types like TempTable might be disabled on the donor and enabled on the client. While this engine won't affect the clone process, it does prevent the clone plugin from running since this check would fail.

Reply:
We could collect clone-supporting plugins, i.e. whose handlertons have `clone_interface.clone_begin != nullptr`